errors.md

```diff
@@ -0,0 +1,156 @@
+# Errors Propagation in ShadowRealm
+
+This document describes the various mechanism that different browsers must implement when exposing message and stack information via Error objects. The objective is to provide guidance for implementers.
+
+## Errors Crossing Callable Boundary
+
+Errors thrown across the ShadowRealm's callable boundary in either direction are replaced by a fresh `TypeError` as described by the spec. Additionally, the new `TypeError` instance can be augmented with a `message` and `stack` properties to help developers.
+
+### New Error.message
+
+The new `TypeError` object created to propagate an error across the callable boundary can be augmented with a `message` property. The value of such property can be a combination of `name` and `message` from the original error. E.g.:
+
+Original `Error.message`:
+
+```
+TypeError: null has no properties
+```
+
+New `Error.message` after crossing a boundary:
+
+```
+Uncaught TypeError: wrapped function threw, error was TypeError: null has no properties
+```
+
+This error allows developers to clearly understand that the error was thrown from another Realm. If the error crosses multiple nested ShadowRealms, the second time the error is created when crossing another boundary, the message should still be formed from scratch rather than providing nesting of the message. The same applies to re-entrancing as well since the error re-entring the ShadowRealm where it was originated, will come as a brand new TypeError object, with no connection to the original Error object.
+
+Accessing the `name` and `message` of the original Error object must not be observed by the user-land code. The following logic may be used:
```

Comment:
@littledan @mhofman my realization was that FF already implemented this logic (I didn't know about it). But it is sound IMO. Please, take a look.